torch/csrc/api/include/torch/nn/functional/loss.h

```diff
@@ -127,6 +127,18 @@ inline Tensor triplet_margin_loss(
       options.reduction());
 }
 
+inline Tensor nll_loss(
+    const Tensor& input,
+    const Tensor& target,
+    const NLLLossOptions& options = {}) {
+  return torch::nll_loss(
+      input,
+      target,
+      options.weight(),
+      options.ignore_index(),
+      options.reduction());
```

Comment:
It would be awesome to implement the equivalent of the Python version:
https://github.com/pytorch/pytorch/blob/783c9c84450f9b75d8cb9752062e961b7449e2c6/torch/nn/functional.py#L1830-L1859
to match the Python version even better :D Thanks so much for your help!

Reply:
We can look at changes to `kl_div`/`mse_loss`/`binary_cross_entropy` as examples in https://github.com/pytorch/pytorch/pull/27942/files

Reply:
Hi @yf225 , would be great if you could provide me some resources which I can study so as to easily translate the code part. I also had a look at the link posted, however, I am not sure about the alternative of `.view or .contiguous()` in C++. It really help me in my future works too.